src/remote/watcher.js

```diff
@@ -34,15 +34,30 @@ export default class RemoteWatcher {
   }
 
   start () {
-    this.watch()
-    this.intervalID = setInterval(this.watch, HEARTBEAT)
+    this.started = new Promise((resolve, reject) => {
+      this.started.resolve = resolve
+      const watch () => {
+        this.watch().catch(err => {
+          reject(err)
```

Comment:
I suppose having `start()` rejecting when first `watch()` does may be useful.
But I would have intuitively assumed that `RemoteWatcher` can be considered "started" as soon as `setInterval()` was called, assuming the first `watch()` call was more related to "polling" / "timing", and not really a part of the "starting" process...
Also, doesn't it make `start()` resolution take some time, since `watch()` may await quite a lot of things?